### PR TITLE
new sequence audio2text-input

### PR DESCRIPTION
**Pull request by piotrek6641**

### CODE REVIEW

**Comment by patuwwy** [comment]
please rebase

File: typescript/audio2text-input/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { Readable } from "stream";
+
+const app: ReadableApp<any> = async function(input: Readable, apiKey: string) {
+    const assembly = new Assembly(apiKey);
+
+    assembly.logger.outputLogStream.pipe(this.logger.inputLogStream);
+
+    return assembly.processFile(input).then(res => {
+        this.logger.info("Result", res);
+        return res;
```
Comment (patuwwy) [inline typescript/audio2text-input/src/index.ts:12]:
```suggestion

        return res;
```

File: typescript/audio2text/src/utils/assembly.ts [hunk 2]
```diff
@@ -3,39 +3,42 @@ import axios, { AxiosInstance } from "axios";
...
-
                 await this.client
                     .post("/upload", data)
                     .then((res) => {
                         this.fileUrl = res.data.upload_url;
-                        resolve("succesfully uploaded");
+                        this.logger.info("succesfully uploaded");
+                        this.logger.info("upload URL", this.fileUrl);
+
+                        resolve()
```
Comment (patuwwy) [inline typescript/audio2text/src/utils/assembly.ts:39]:
```suggestion
                        resolve();
```

File: typescript/audio2text/src/utils/assembly.ts [hunk 3]
```diff
@@ -46,30 +49,45 @@ export class Assembly {
...
+                    this.logger.info("Got error from API", res.data.error);
+                }
+
+                return res.data;
+            });
+
+        if (result.status !== "completed") {
+            this.logger.info(result.status);
+            await defer(3000);
+            return this.transcript();
```
Comment (patuwwy) [inline typescript/audio2text/src/utils/assembly.ts:77]:
```suggestion

            return this.transcript();
```

File: typescript/audio2text/src/utils/assembly.ts [hunk 4]
```diff
@@ -46,30 +49,45 @@ export class Assembly {
...
+        }
+
+        return result.text;
+    }
+    async processFileFs(): Promise<string> {
+        await this.uploadFromFs();
+        await this.startTranscript();
+        return this.transcript();
```
Comment (patuwwy) [inline typescript/audio2text/src/utils/assembly.ts:85]:
```suggestion

        return this.transcript();
```

File: typescript/audio2text/src/utils/assembly.ts [hunk 5]
```diff
@@ -3,39 +3,42 @@ import axios, { AxiosInstance } from "axios";
 import { defer } from "@scramjet/utility";
 import { readFile } from "fs";
 import path from "path";
+import { ObjLogger } from "@scramjet/obj-logger";
+import { Readable } from "stream";
 
 
 export class Assembly {
     //path to file change song.wav to name of your file
```
Comment (patuwwy) [inline typescript/audio2text/src/utils/assembly.ts:11]: filename could be provided as sequence parameter

**Review by alicja-gruzdz — APPROVED** [review]

**Pull request merged**